### PR TITLE
Fix OSS CI macos test-custom-ops-macos

### PR DESCRIPTION
**Pull request by kirklandsign**

### CODE REVIEW

**Comment by kirklandsign** [comment]
Got error https://productionresultssa1.blob.core.windows.net/actions-results/89456ed1-b839-4e2b-975a-f3b989fcd7b9/workflow-job-run-ca7d7c50-4682-5db1-d3ca-b614f46f4c5f/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-07-08T21%3A28%3A25Z&sig=DAj0M5KXQ%2FGO4fTaaJAzu2NQKvRCTvHM5jgeFU%2FwuJs%3D&ske=2024-07-09T05%3A30%3A49Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-07-08T17%3A30%3A49Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2023-11-03&sp=r&spr=https&sr=b&st=2024-07-08T21%3A18%3A20Z&sv=2023-11-03

```
2024-07-08T21:17:39.6493720Z make[2]: *** No rule to make target `LIB_executorch_no_prim_ops_shared-NOTFOUND', needed by `libcustom_ops_aot_lib.dylib'.  Stop.
2024-07-08T21:17:39.6494290Z make[2]: *** Waiting for unfinished jobs....
```

**Comment by kirklandsign** [comment]
```
Exporting custom_ops_2.pte
No registered custom op my_ops::mul4.out
Traceback (most recent call last):
  File "/Users/hsz/miniconda3/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/hsz/miniconda3/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/hsz/executorch/examples/portable/custom_ops/custom_ops_2.py", line 56, in <module>
    torch.ops.load_library(args.so_library)
  File "/Users/hsz/miniconda3/lib/python3.10/site-packages/torch/_ops.py", line 1048, in load_library
    ctypes.CDLL(path)
  File "/Users/hsz/miniconda3/lib/python3.10/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(/Users/hsz/executorch/cmake-out/examples/portable/custom_ops/libcustom_ops_aot_lib.dylib, 0x0006): Library not loaded: @rpath/libexecutorch_no_prim_ops_shared.dylib
  Referenced from: <18EEFCB1-471E-3C05-9F11-BE9D8A7BB703> /Users/hsz/executorch/cmake-out/examples/portable/custom_ops/libcustom_ops_aot_lib.dylib
  Reason: tried: '/Users/hsz/miniconda3/lib/python3.10/site-packages/torch/lib/libexecutorch_no_prim_ops_shared.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/hsz/miniconda3/lib/python3.10/site-packages/torch/lib/libexecutorch_no_prim_ops_shared.dylib' (no such file), '/Users/hsz/miniconda3/lib/python3.10/site-packages/torch/lib/libexecutorch_no_prim_ops_shared.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/hsz/miniconda3/lib/python3.10/site-packages/torch/lib/libexecutorch_no_prim_ops_shared.dylib' (no such file), '/Users/hsz/miniconda3/lib/python3.10/lib-dynload/../../libexecutorch_no_prim_ops_shared.dylib' (no such file), '/Users/hsz/miniconda3/bin/../lib/libexecutorch_no_prim_ops_shared.dylib' (no such file)
```

File: examples/portable/custom_ops/test_custom_ops.sh [hunk 1]
```diff
@@ -103,7 +103,7 @@ test_cmake_custom_op_2() {
 
   EXT=$(get_shared_lib_ext)
   echo "Exporting ${model_name}.pte"
-  ${PYTHON_EXECUTABLE} -m "examples.portable.custom_ops.${model_name}" --so_library="cmake-out/examples/portable/custom_ops/libcustom_ops_aot_lib$EXT"
+  DYLD_LIBRARY_PATH="cmake-out/lib" ${PYTHON_EXECUTABLE} -m "examples.portable.custom_ops.${model_name}" --so_library="cmake-out/examples/portable/custom_ops/libcustom_ops_aot_lib$EXT"
```
Comment (larryliu0820) [inline examples/portable/custom_ops/test_custom_ops.sh:106]: Why is this necessary?

Comment (kirklandsign) [reply examples/portable/custom_ops/test_custom_ops.sh:106]: https://github.com/pytorch/executorch/actions/runs/9847259938/job/27186835771 can’t find dl path

**Review by larryliu0820 — APPROVED** [review]
